bb-pilcom/bb-pil-backend/templates/recursive_verifier.cpp.hbs

```diff
@@ -0,0 +1,158 @@
+    const StdlibProof<Builder>& stdlib_proof,
+    const std::vector<std::vector<FF>>& public_inputs,
+    AggregationObject agg_obj)
+{
+    using Curve = typename Flavor::Curve;
+    using Zeromorph = ZeroMorphVerifier_<Curve>;
+    using PCS = typename Flavor::PCS;
+    using VerifierCommitments = typename Flavor::VerifierCommitments;
+    using CommitmentLabels = typename Flavor::CommitmentLabels;
+    using RelationParams = ::bb::RelationParameters<typename Flavor::FF>;
+    using Transcript = typename Flavor::Transcript;
+
+    transcript = std::make_shared<Transcript>(stdlib_proof);
+
+    RelationParams relation_parameters;
+    VerifierCommitments commitments{ key };
+    CommitmentLabels commitment_labels;
+
+    const auto circuit_size = transcript->template receive_from_prover<FF>("circuit_size");
+    if (static_cast<uint32_t>(circuit_size.get_value()) != key->circuit_size) {
+        throw_or_abort("{{name}}RecursiveVerifier::verify_proof: proof circuit size does not match verification key!");
+    }
+
+    // Get commitments to VM wires
+    for (auto [comm, label] : zip_view(commitments.get_wires(), commitment_labels.get_wires())) {
+        comm = transcript->template receive_from_prover<Commitment>(label);
+    }
+
+    auto [beta, gamma] = transcript->template get_challenges<FF>("beta", "gamma");
+    relation_parameters.beta = beta;
+    relation_parameters.gamma = gamma;
+
+    // Get commitments to inverses
+    for (auto [label, commitment] : zip_view(commitment_labels.get_derived(), commitments.get_derived())) {
+        commitment = transcript->template receive_from_prover<Commitment>(label);
+    }
+
+    // unconstrained
+    const size_t log_circuit_size = numeric::get_msb(static_cast<uint32_t>(circuit_size.get_value()));
+    auto sumcheck = SumcheckVerifier<Flavor>(log_circuit_size, transcript);
+
+    FF alpha = transcript->template get_challenge<FF>("Sumcheck:alpha");
+
+    auto gate_challenges = std::vector<FF>(log_circuit_size);
+    for (size_t idx = 0; idx < log_circuit_size; idx++) {
+        gate_challenges[idx] = transcript->template get_challenge<FF>("Sumcheck:gate_challenge_" + std::to_string(idx));
+    }
+
+    // No need to constrain that sumcheck_verified is true as this is guaranteed by the implementation of
+    // when called over a "circuit field" types.
+    auto [multivariate_challenge, claimed_evaluations, sumcheck_verified] =
+        sumcheck.verify(relation_parameters, alpha, gate_challenges);
+
+    vinfo("verified sumcheck: ", (sumcheck_verified.has_value() && sumcheck_verified.value()));
+
+    // Public columns evaluation checks
+    std::vector<FF> mle_challenge(multivariate_challenge.begin(),
+                                  multivariate_challenge.begin() + static_cast<int>(log_circuit_size));
+
+    {{#each public_cols}}
+    FF {{col}}_evaluation = evaluate_public_input_column(public_inputs[{{idx}}], mle_challenge);
```

Comment:
Can't this be extracted as a function and be the only code-generated portion?

Reply:
Potentially yes, but in the rest of the template we also codegen the prefix Avm ({{name}}). I think verifier.cpp has almost the same structure.

Reply:
Yes I'd rathe follow whatever we already do for the sake of uniformity, then we can consider changing things.